internal/native_image/classic.bzl

```diff
@@ -0,0 +1,88 @@
+"Legacy ('classic') rules for building with GraalVM on Bazel."
+
+load(
+    "//internal/native_image:common.bzl",
+    _BAZEL_CPP_TOOLCHAIN_TYPE = "BAZEL_CPP_TOOLCHAIN_TYPE",
+    _BAZEL_CURRENT_CPP_TOOLCHAIN = "BAZEL_CURRENT_CPP_TOOLCHAIN",
+    _DEFAULT_GVM_REPO = "DEFAULT_GVM_REPO",
+    _GVM_TOOLCHAIN_TYPE = "GVM_TOOLCHAIN_TYPE",
+    _NATIVE_IMAGE_ATTRS = "NATIVE_IMAGE_ATTRS",
+    _RULES_REPO = "RULES_REPO",
+    _prepare_native_image_rule_context = "prepare_native_image_rule_context",
+)
+load(
+    "//internal/native_image:toolchain.bzl",
+    _resolve_cc_toolchain = "resolve_cc_toolchain",
+)
+
+def _graal_binary_classic_implementation(ctx):
+    graal_attr = ctx.attr.native_image_tool
```

Comment:
legacy rules have now been split into their own "classic" implementation, which is an entirely different rule (instead of combing both paths under one rule with conditionals).

this simplifies both rule implementations and obviates the need for the `_legacy_rule` attr